### ISSUE TITLE
[jsx-curly-spacing] False positive with render props

### ISSUE THREAD

**Issue by FredyC** [palantir/tslint-react#140]
The following code is causing false positive on this rule.

**tslint-react: 3.4.0**

```jsx
<Route
  children={({ location, history }) =>
    store.auth.isAuthenticated ? (
      <Redirect to={location.state.from} />
    ) : null
  } // Error here: There should be no space before } (jsx-curly-spacing)
/>
```

I might add that code is auto formatted by Prettier so I cannot write it differently :)

/cc @bolatovumar

**Comment by adidahiya** [comment]
There are many tslint rules which are incompatible with prettier. I suggest you use `tslint-config-prettier` to avoid any conflicts.

**Comment by FredyC** [comment]
@adidahiya Well, it was merely a side note that I am using Prettier :) Imagine if someone would format their code this way manually (there is nothing wrong about such format) and get the same error. In that case it's obviously a bug in the rule, don't you agree?

**Comment by adidahiya** [comment]
sure, I guess so... would accept PRs to fix this false positive. typically the rules in `tslint-react` are all used together, and that code wouldn't pass `jsx-no-multiline-js` anyway...

**Comment by peter50216** [comment]
Hi,
Can a new minor version be released that include this fix? Thanks.

**Comment by fqlx** [comment]
+1, seeing this issue as well.

**Comment by evenfrost** [comment]
Same here.

**Comment by bartvde** [comment]
Can we please have a release that fixes this?
